newrelic/api/message_trace.py

```diff
@@ -145,7 +175,16 @@ def _nr_message_trace_wrapper_(wrapped, instance, args, kwargs):
     return FunctionWrapper(wrapped, _nr_message_trace_wrapper_)
 
 
-def message_trace(library, operation, destination_type, destination_name, params={}, terminal=True, async_wrapper=None):
+def message_trace(
+    library,
+    operation,
+    destination_type,
+    destination_name,
+    params={},
+    terminal=True,
+    async_wrapper=None,
+    extract_agent_attrs=None,
```

Comment:
I don't particularly care for muddying the public APIs with this. Can we instead define an SQS wrapper in the botocore instrumentation that starts a message trace, and adds the necessary attrs?

Reply:
I've recreated a simpler wrapper inside the botocore instrumentation that contains the call to create the agent attributes.